api/casauth.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"github.com/wtg/shuttletracker/database"
+	"gopkg.in/cas.v2"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// CasClient stores the local cas client and an instance of the database
+type CasClient struct {
+	cas *cas.Client
+	db  database.Database
+}
+
+// Create creates a new CasClient from a casurl and a database
+func (cli *CasClient) Create(url *url.URL, db database.Database) {
```

Comment:
This is not quite idiomatic Go. Normally we'd do a `NewCASClient` function or something, but I don't even think that is necessary here. The user (api.go) can just instantiate the struct directly with the fields.